src/azul/transformer.py

```diff
@@ -300,6 +300,43 @@ def get(self):
         return thaw(super().get())
 
 
+class FrequencySetAccumulator(Accumulator):
+    """
+    An accumulator that accepts any number of values and returns a list that has
+    length max_size or smaller and is ordered by most frequent value to less frequent.
+
+    >>> a = FrequencySetAccumulator(2)
+    >>> a.accumulate('x')
+    True
+    >>> a.accumulate(['x','y'])
+    True
+    >>> a.accumulate({'x','y','z'})
+    True
+    >>> a.get()
+    ['x', 'y']
+    >>> a = FrequencySetAccumulator(0)
+    >>> a.accumulate('x')
+    True
+    >>> a.get()
+    []
+    """
+    def __init__(self, max_size) -> None:
+        super().__init__()
+        self.value = Counter()
+        self.max_size = max_size
+
+    def accumulate(self, value) -> bool:
+        original_count = sum(self.value.values())
+        if isinstance(value, (list, set)):
+            self.value.update(value)
+        else:
+            self.value[value] += 1
+        return original_count < sum(self.value.values())
+
+    def get(self) -> List[Any]:
+        return [item for item, count in self.value.most_common(self.max_size)]
```

Comment:
```suggestion
        return {item for item, count in self.value.most_common(self.max_size)}
```

Note that this might make the assertions in the doctest a little more tricky. I do think the type of the return value should reflect the uniqueness property even though that discards the ordering.